Message forwarding (#6833)
hammerandchisel/discord-api-docs: docs/Change_Log.md

# Change Log

## Message Forwarding rollout

#### July 15, 2024

> We are slowly rolling out the message forwarding feature to users. This feature allows callers to create a message using `message_reference.type = FORWARD` and have the API generate a `message_snapshot` for the sent message. The feature some limitations (see documentation) and the snapshot is a minimial version of a standard `MessageObject`, but does capture the core parts of a message.

Reviewer: ```suggestion
We are slowly rolling out the message forwarding feature to users. This feature allows callers to create a message using `message_reference.type = FORWARD` and have the API generate a `message_snapshot` for the sent message. The feature has [some limitations](#DOCS_RESOURCES_CHANNEL/message-reference-types) and the snapshot is a minimal version of a standard `MessageObject`, but does capture the core parts of a message.
```

> We are slowly rolling out the message forwarding feature to users. This feature allows callers to create a message using `message_reference.type = FORWARD` and have the API generate a `message_snapshot` for the sent message. The feature has [some limitations](#DOCS_RESOURCES_CHANNEL/message-reference-types) and the snapshot is a minimal version of a standard `MessageObject`, but does capture the core parts of a message.

Reviewer 2: ```suggestion
We are slowly rolling out the message forwarding feature to users. This feature allows to create a message using `message_reference.type = FORWARD` and have the API generate a `message_snapshot` for the sent message. The feature has [some limitations](#DOCS_RESOURCES_CHANNEL/message-reference-types) and the snapshot is a minimal version of a standard message object that only captures the core parts.
```

The resulting message will look something like:
```json
{

>   "id": 1234,

Reviewer 3: ```suggestion
  "id": "1255957733279273083",
```
more realistic id as a string

  "message_reference": {
    "type": 1, // Forward
    ...
  }
  "message_snapshots": [
    {
      "message": {
        "content": "original message",
        "embeds": [...],
        "attachments": [...],
        ...
      }
    }
  ],
  ...
}
```

> We have applied stricter rate limits for this feature based on (intentionally omitting values since they might change as needed):

Reviewer: ```suggestion
We have applied stricter rate limits for this feature based on the following:
```

> - number of forwards sent

Reviewer 2: I think this could use some clarification: Does this mean how often this message has already been forwarded globally? Or forwards per user (of this message)?

Reviewer 4: forwards per user

Reviewer 4: it is the # of forwards sent by the user. So each user can send 5 forwards in 5 secs. Or something like that.

- total attachment size

###### API Updates since preview

> This was [previously announced](https://canary.discord.com/channels/613425648685547541/1233463756160503859) and has since had a slight API changes depeneding on when you last read/used the api:

Reviewer: ```suggestion
This was [previously announced](https://discord.com/channels/613425648685547541/697138785317814292/1233463756160503859) but note that the final API has a few changes since the API was first previewed:
```

Reviewer 5: looks like this links to the thread, IMO we should link to the announcement message instead

https://discord.com/channels/613425648685547541/697138785317814292/1233463756160503859

Reviewer: updated! thanks @advaith1

> - `message_snapshot.guild` is not returned anymore
> - `message_reference` is now populated with `type == FORWARD`

Reviewer 2: ```suggestion
- [`message snapshot`](#DOCS_RESOURCES_CHANNEL/message-snapshot-object) objects don't include a `guild` field anymore since the `message_reference` already provides that information
- forwarded messages have a distinctive `message_reference` type of `FORWARD` now
```
